render/wgpu/src/backend.rs

```diff
@@ -51,17 +51,25 @@ impl WgpuRenderBackend<SwapChainTarget> {
     ) -> Result<Self, Error> {
         let instance = wgpu::Instance::new(wgpu::Backends::BROWSER_WEBGPU | wgpu::Backends::GL);
         let surface = instance.create_surface_from_canvas(canvas)?;
-        let descriptors = Self::build_descriptors(
-            wgpu::Backends::BROWSER_WEBGPU | wgpu::Backends::GL,
-            instance,
-            Some(&surface),
-            wgpu::PowerPreference::HighPerformance,
-            None,
+        let adapter = instance
+            .request_adapter(&wgpu::RequestAdapterOptions {
+                power_preference: wgpu::PowerPreference::HighPerformance,
+                compatible_surface: Some(&surface),
+                force_fallback_adapter: false,
+            })
+            .await
+            .ok_or_else(|| "Unsupported device".to_string())?;
```

Comment:
This makes us lose the nicer error message from `build_descriptors`. I think it would be better to better to call `detect_buffer_bug` from `build_descriptors` (possibly behind a flag).

Reply:
I'll split up build_descriptors into the adapter finding stuff only, we need that anyway for the upcoming libretro stuff

Reply:
Actually no nvm that means multiple `block_ons` at the caller, that can be a later change. I'll add a param!

Reply:
Ended up with cfg'ing it since it's only for web